src/apps/packet_filter/README.md.src

```diff
@@ -54,15 +74,11 @@ is allowed.
 
 — Key **state_track**
 
-*Optional*. Tracks connections state in a named connection table.  If used
-inside a rule, every packet that passes this specific rule is tracked.
-If used 'outside' any rule, packets passing _any_ rule on this app will be
-tracked.
+*Optional*. A string naming a connection table. If set, packets passing
+the rule are tracked in the specified connection table.
 
 — Key **state_check**
 
-*Optional*. Checks if a packet belongs to a connection in a named
-connection table.  If used inside a rule, a packet must belong to an
-existing connection in addition to any other condition in the rule to pass.
-If used 'outside' any rule, if a packet belongs to an existing connection
-it's enough to pass the app.
+*Optional*. A string denoting a connection table. If set, packets must
```

Comment:
That sounds reasonable. Thanks for explaining. If the PacketFilter has an API that makes sense and is accurately documented then that is excellent.

I am also thinking about the effect of changes to one part of the code base on other parts. The "packet filtering ecosystem" that we have today looks like this:

- PacketFilter app.
- snabbnfv-traffic process (uses and configures PacketFilter app).
- neutron2snabb program (translates Neutron configuration to snabbnfv-traffic).
- [Neutron API Extensions](https://github.com/SnabbCo/snabb-nfv/wiki/Neutron-API-Extensions) where we need a section explaining our interpretation of Neutron Security Groups.

So Max and I need to describe our Neutron configuration interface and make sure that it is correctly mapped onto the PacketFilter app.